src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.cpp

```diff
@@ -363,11 +372,22 @@ void TlsCertificateManagementServer::HandleProvisionClientCertificate(HandlerCon
     VerifyOrReturn(req.ccdid <= kMaxClientCertId, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
     VerifyOrReturn(req.clientCertificate.size() <= kSpecMaxCertBytes,
                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+    ReturnOnFailure(Crypto::IsCertificateValidAtCurrentTime(req.clientCertificate),
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::DynamicConstraintError));
     size_t intermediateSize;
-    VerifyOrReturn(req.intermediateCertificates.ComputeSize(&intermediateSize) == CHIP_NO_ERROR,
-                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand));
+    ReturnOnFailure(req.intermediateCertificates.ComputeSize(&intermediateSize),
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand));
     VerifyOrReturn(intermediateSize <= kMaxIntermediateCertificates,
                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+    auto srcIter = req.intermediateCertificates.begin();
+    while (srcIter.Next())
+    {
+        auto & cert = srcIter.GetValue();
+        VerifyOrReturn(cert.size() <= kSpecMaxCertBytes, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+
+        ReturnOnFailure(Crypto::IsCertificateValidAtCurrentTime(cert),
+                        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::DynamicConstraintError));
+    }
 
```

Comment:
Where's the TLV validity check for req.intermediateCertificates?  As in, what if one of those list entries fails to decode entirely?  This will silently truncate the list at that point.

Reply:
That's what this is; we're in a while loop of `srcIter = req.intermediateCertificates`, where `srcIter.GetValue()` - if it's an invalid cert (i.e. fails to decode), `Crypto::IsCertificateValidAtCurrentTime` will fail because it can't check if the cert is at a valid time unless it's a valid cert

Reply:
Addressed @ https://github.com/project-chip/connectedhomeip/pull/41046 - thanks again for catching this